src/Microsoft.Data/PrimitiveColumnContainer.cs

```diff
@@ -75,8 +75,47 @@ public PrimitiveColumnContainer(IEnumerable<T?> values)
 
         public PrimitiveColumnContainer(ReadOnlyMemory<byte> buffer, ReadOnlyMemory<byte> nullBitMap, int length, int nullCount)
         {
-            Buffers.Add(new ReadOnlyDataFrameBuffer<T>(buffer, length));
-            NullBitMapBuffers.Add(new ReadOnlyDataFrameBuffer<byte>(nullBitMap, length));
+            ReadOnlyDataFrameBuffer<T> dataBuffer;
+            if (buffer.IsEmpty)
+            {
+                DataFrameBuffer<T> mutableBuffer = new DataFrameBuffer<T>();
+                mutableBuffer.EnsureCapacity(length);
+                mutableBuffer.Length = length;
+                mutableBuffer.RawSpan.Fill(default(T));
+                dataBuffer = mutableBuffer;
+            }
+            else
+            {
+                dataBuffer = new ReadOnlyDataFrameBuffer<T>(buffer, length);
+            }
+            Buffers.Add(dataBuffer);
+            int bitMapBufferLength = (length + 7) / 8;
+            ReadOnlyDataFrameBuffer<byte> nullDataFrameBuffer;
+            if (nullBitMap.IsEmpty)
+            {
+                if (nullCount != 0)
+                {
+                    throw new ArgumentNullException(Strings.InconsistentNullBitMapAndNullCount, nameof(nullBitMap));
+                }
+                if (!buffer.IsEmpty)
+                {
+                    // Create a new bitMap with all the bits upto length set
+                    var bitMap = new byte[bitMapBufferLength];
+                    bitMap.AsSpan().Slice(0, bitMapBufferLength - 1).Fill(255);
+                    int lastByte = 1 << (length - (bitMapBufferLength - 1) * 8); 
+                    bitMap[bitMapBufferLength - 1] = (byte)(lastByte - 1);
+                    nullDataFrameBuffer = new DataFrameBuffer<byte>(bitMap, bitMapBufferLength);
+                }
+                else
+                {
+                    nullDataFrameBuffer = new DataFrameBuffer<byte>();
+                }
+            }
+            else
+            {
+                nullDataFrameBuffer = new ReadOnlyDataFrameBuffer<byte>(nullBitMap, bitMapBufferLength);
```

Comment:
Should we check if `nullBitMap` is at least `bitMapBufferLength` long?

Reply:
Yup. Good idea. Added a unit test too.